ZividOpenCV/CaptureUndistortRGB/CaptureUndistortRGB.cpp

```diff
@@ -0,0 +1,129 @@
+
+        cv::Size size = rgb.size();
+
+        cv::Mat rgbUndistorted;
+        cv::Mat rgbUndistortedFull;
+
+        cv::Mat distortionCoefficients(cv::Size(1, 5), CV_64FC1, cv::Scalar(0));
+        cv::Mat cameraMatrix(cv::Size(3, 3), CV_64FC1, cv::Scalar(0));
+        cv::Mat optimalCameraMatrix;
+
+        getCameraIntrinsics(camera, distortionCoefficients, cameraMatrix);
+
+        optimalCameraMatrix = cv::getOptimalNewCameraMatrix(cameraMatrix, distortionCoefficients, size, 1, size);
+
+        cv::undistort(rgb, rgbUndistorted, cameraMatrix, distortionCoefficients);
+        cv::undistort(rgb, rgbUndistortedFull, cameraMatrix, distortionCoefficients, optimalCameraMatrix);
+
+        std::cout << "Displaying and saving the RGB image" << std::endl;
+
+        displayRGB(rgb, "Distorted RGB image");
+        cv::imwrite("Distorted RGB image.jpg", rgb);
+
+        std::cout << "Displaying and saving the undistorted RGB image" << std::endl;
+
+        displayRGB(rgbUndistorted, "Undistorted RGB image");
+        cv::imwrite("Undistorted RGB image.jpg", rgbUndistorted);
+
+        std::cout << "Displaying and saving the Undistorted RGB image - full" << std::endl;
+
+        displayRGB(rgbUndistortedFull, "Undistorted RGB image - full");
+        cv::imwrite("Undistorted RGB image - full.jpg", rgbUndistorted);
+    }
+    catch(const std::exception &e)
+    {
+        std::cerr << "Error: " << Zivid::toString(e) << std::endl;
+        return EXIT_FAILURE;
+    }
+}
+
+void pointCloudToRGB(const Zivid::PointCloud &pointCloud, cv::Mat &rgb)
+{
+    for(int i = 0; i < pointCloud.height(); i++)
+    {
+        for(int j = 0; j < pointCloud.width(); j++)
+        {
+            cv::Vec3b &color = rgb.at<cv::Vec3b>(i, j);
+            color[0] = pointCloud(i, j).blue();
+            color[1] = pointCloud(i, j).green();
+            color[2] = pointCloud(i, j).red();
+        }
+    }
+}
+
+void getCameraIntrinsics(const Zivid::Camera &camera, cv::Mat &distortionCoefficients, cv::Mat &cameraMatrix)
+{
+    distortionCoefficients.at<double>(0, 0) = camera.intrinsics().distortion().k1().value();
+    distortionCoefficients.at<double>(0, 1) = camera.intrinsics().distortion().k2().value();
+    distortionCoefficients.at<double>(0, 2) = camera.intrinsics().distortion().p1().value();
+    distortionCoefficients.at<double>(0, 3) = camera.intrinsics().distortion().p2().value();
+    distortionCoefficients.at<double>(0, 4) = camera.intrinsics().distortion().k3().value();
+
+    cameraMatrix.at<double>(0, 0) = camera.intrinsics().cameraMatrix().fx().value();
+    cameraMatrix.at<double>(0, 2) = camera.intrinsics().cameraMatrix().cx().value();
+    cameraMatrix.at<double>(1, 1) = camera.intrinsics().cameraMatrix().fy().value();
+    cameraMatrix.at<double>(1, 2) = camera.intrinsics().cameraMatrix().cy().value();
+    cameraMatrix.at<double>(2, 2) = 1;
+}
+
+void displayRGB(const cv::Mat &rgb, const std::string rgbName)
```

Comment:
`rgbName` can be a reference. It won't matter performancewise, but it's just less surprising to read.

Reply:
Done